src/app/modules/admin/component/employees/attach-user/attach-user.component.ts

```diff
@@ -0,0 +1,114 @@
+import { Component, EventEmitter, Output, OnInit, Input } from '@angular/core';
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+import { ToastrService } from 'ngx-toastr';
+import { NAME_FIELD_ERRORS } from 'src/app/custom-errors';
+import { Employee } from 'src/app/modules/shared/models/employee';
+import { Post } from 'src/app/modules/shared/models/post';
+import { EmployeeService } from 'src/app/modules/shared/services/employee.service';
+import { PostService } from 'src/app/modules/shared/services/post.service';
+import { UniqueFieldValidator } from 'src/app/modules/shared/validators/unique-field-validator';
+import { UserService } from 'src/app/modules/shared/services/user.service';
+import { User } from 'src/app/modules/shared/models/user';
+import { HttpErrorResponse } from '@angular/common/http';
+import { SpinnerService } from 'src/app/modules/core/services/spinner.service';
+import { map, catchError } from 'rxjs/operators';
+import { Observable, throwError } from 'rxjs';
+
+@Component({
+    selector: 'app-attach-user',
+    templateUrl: './attach-user.component.html',
+    styleUrls: ['./attach-user.component.scss']
+})
+
+export class AttachUserComponent implements OnInit {
+
+    constructor(
+        protected progress: SpinnerService,
+        private employeeService: EmployeeService,
+        private userService: UserService,
+        private toast: ToastrService
```

Comment:
please use naming exact like service name but camlecase, eg spinnerService instead of progress